sdk/keyvault/azure-security-keyvault/tests/test_examples_keys.py

```diff
@@ -0,0 +1,297 @@
+                print(key.version)
+                print(key.key_material.kty)
+
+            # [END list_key_versions]
+        except HttpResponseError:
+            pass
+
+        try:
+            # [START list_deleted_keys]
+
+            # get a list of deleted keys (requires soft-delete enabled for the vault)
+            deleted_keys = key_client.list_deleted_keys()
+
+            for key in deleted_keys:
+                print(key.id)
+                print(key.name)
+
+            # [END list_deleted_keys]
+        except HttpResponseError:
+            pass
+
+    @ResourceGroupPreparer()
+    @VaultClientPreparer()
+    def test_example_keys_backup_restore(self, vault_client, **kwargs):
+        key_client = vault_client.keys
+        created_key = key_client.create_key("keyrec", "RSA")
+        key_name = created_key.name
+        try:
+            # [START backup_key]
+            # backup key
+            key_backup = key_client.backup_key(key_name)
+
+            # returns the raw bytes of the backed up key
+            print(key_backup)
+
+            # [END backup_key]
+
+            key_client.delete_key("keyrec")
+            if self.is_live:
+                # wait a second to ensure the key has been deleted
+                time.sleep(20)
+            # [START restore_key]
+
+            # restores a backed up key
+            restored_key = key_client.restore_key(key_backup)
+            print(restored_key.id)
+            print(restored_key.version)
+
+            # [END restore_key]
+        except ResourceExistsError:
+            pass
+
+    @ResourceGroupPreparer()
+    @VaultClientPreparer(enable_soft_delete=True)
+    def test_example_keys_recover_purge(self, vault_client, **kwargs):
+        key_client = vault_client.keys
+        created_key = key_client.create_key("key-name", "RSA")
+        if self.is_live:
+            # wait a second to ensure the key has been created
+            time.sleep(20)
+        key_client.delete_key(created_key.name)
+        if self.is_live:
+            # wait a second to ensure the key has been deleted
+            time.sleep(30)
+
+        try:
+            # [START get_deleted_key]
+            # gets a deleted key (requires soft-delete enabled for the vault)
+            deleted_key = key_client.get_deleted_key("key-name")
+            print(deleted_key.name)
+
+            # [END get_deleted_key]
+        except ResourceNotFoundError:
+            pass
+
+        try:
+            # [START recover_deleted_key]
+
+            # recover deleted key to its latest version
+            recover_deleted_key = key_client.recover_deleted_key("key-name")
+            print(recover_deleted_key.id)
+            print(recover_deleted_key.name)
+
+            # [END recover_deleted_key]
+        except HttpResponseError:
+            pass
+
+        try:
+            if self.is_live:
+                # wait a second to ensure the key has been recovered
+                time.sleep(20)
```

Comment:
Is this necessary? Recover returns the recovered key, so in synchronous code the key should be deletable immediately afterward.